### PR TITLE
Fixed https://github.com/Ashald/EnvFile/issues/27

### LINKED ISSUE #27: Environment variables aren't loaded in IDEA
Hi,

First of all, thank you for this nice plugin ! 

I installed it but it didn't load the environment variables with IntelliJ IDEA Community Edition 2016.3.5. For information:
- Here is the run configuration:  settings
- The .env file just contains `FOO=BAR`
- `System.getenv().containsKey("FOO")` returns `false`

Could you please tell me if I missed something ?

Yoan

### PR DESCRIPTION
**Pull request by yoanthiebault**
This commit aims to fix https://github.com/Ashald/EnvFile/issues/27. The idea is to use `updateJavaParameters` instead of `patchCommandLine` in `IdeaRunConfigurationExtension`

### CODE REVIEW

**Comment by Ashald** [comment]
Thanks for the contribution! I looked over the code and it looks good to me generally, although I would prefer to change one thing in terms of the interface.

File: project/platform/src/net/ashald/envfile/platform/ui/EnvFileConfigurationEditor.java [hunk 1]
```diff
@@ -108,10 +109,16 @@ public static void writeExternal(@NotNull RunConfigurationBase configuration, @N
     }
 
     public static void patchCommandLine(@NotNull RunConfigurationBase configuration, @Nullable RunnerSettings runnerSettings, @NotNull GeneralCommandLine cmdLine) throws ExecutionException {
-        EnvFileSettings state = configuration.getUserData(USER_DATA_KEY);
-        if (state != null && state.isEnabled()) {
-            Map<String, String> source = cmdLine.getEnvironment();
+        patchEnvironmentVariables(configuration, cmdLine.getEnvironment());
+    }
 
+    public static void patchJavaParameters(@NotNull RunConfigurationBase configuration, JavaParameters params, RunnerSettings runnerSettings) throws ExecutionException {
```
Comment (ashald) [inline project/platform/src/net/ashald/envfile/platform/ui/EnvFileConfigurationEditor.java]: The initial design aim for this class was to keep it as generic as possible and use it in concrete implementations. From this perspective rather then keeping methods like `patchCommandLine`  and `patchJavaParameters` I'd rather go with single method `patchEnvironmentVariables` (as you essentially did) and move 1-liners from `patchCommandLine`/`patchJavaParameters` to concrete implementations.

**Review by ashald — CHANGES_REQUESTED** [review]
Thanks for the contribution! I looked over the code and it looks good to me generally, although I would prefer to change one thing in terms of the interface.

File: project/platform/src/net/ashald/envfile/platform/ui/EnvFileConfigurationEditor.java (hunk 1 above)
Comment (yoanthiebault) [reply project/platform/src/net/ashald/envfile/platform/ui/EnvFileConfigurationEditor.java]: Done !
Just for information, I couldn't test it with RubyMine and PyCharm but it should work :)

**Review by ashald — APPROVED** [review]
Looks good!

**Pull request merged**